File: CS/CS.NET-VB.NET/ASP.NET 4.0 Book/Source/Chapter 18/VB/ConnectionZoneAppVB/Provider.ascx.vb

Partial Class Provider
    Inherits System.Web.UI.UserControl
    Implements IDataTransmit

    <WebControls.WebParts.ConnectionProvider("Text", "TextProvider")> _
    Public Function RetriveDataInterface() As IDataTransmit
        Return CType(Me, IDataTransmit)
    End Function

    Public Function RetriveData() As String Implements IDataTransmit.RetriveData
        Return TextBox1.Text
    End Function

End Class

File: CS/CS.NET-VB.NET/ASP.NET 4.0 Book/Source/Chapter 18/CS/ConnectionZoneAppCS/Provider.ascx.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;

public partial class Provider : System.Web.UI.UserControl, IDataTransmit
{
    [ConnectionProvider("Text", "TextProvider")]
    public IDataTransmit RetriveDataInterface()
    {
        return (IDataTransmit)this;
    }
    public string RetriveData()
    {
        return TextBox1.Text;
    }

}